packages/odyssey-react-mui/src/OdysseyThemeProvider.tsx

```diff
@@ -29,48 +83,108 @@ export type OdysseyThemeProviderProps = {
+  ({
+    children,
+    designTokensOverride: designTokensOverrideProp,
+    shadowDomElement: shadowDomElementProp,
+    shadowRootElement: shadowRootElementProp,
+    themeOverride: themeOverrideProp,
+    contrastMode: explicitContrastMode,
+  }: OdysseyThemeProviderProps) => {
+    const odysseyThemeProviderPropsContext = useContext(
+      OdysseyThemeProviderPropsContext,
+    );
 
-  const odysseyTheme = useMemo(
-    () =>
-      createOdysseyMuiTheme({
-        odysseyTokens,
-        shadowRootElement: shadowRootElement || shadowDomElement,
+    const computedProps = useMemo(
+      () => ({
+        designTokensOverride:
+          designTokensOverrideProp ??
+          odysseyThemeProviderPropsContext.designTokensOverride,
+        shadowDomElement:
+          shadowDomElementProp ??
+          odysseyThemeProviderPropsContext.shadowDomElement,
+        shadowRootElement:
+          shadowRootElementProp ??
+          odysseyThemeProviderPropsContext.shadowRootElement,
+        themeOverride:
+          themeOverrideProp ?? odysseyThemeProviderPropsContext.themeOverride,
       }),
-    [odysseyTokens, shadowDomElement, shadowRootElement],
-  );
+      [
+        designTokensOverrideProp,
+        odysseyThemeProviderPropsContext.designTokensOverride,
+        odysseyThemeProviderPropsContext.shadowDomElement,
+        odysseyThemeProviderPropsContext.shadowRootElement,
+        odysseyThemeProviderPropsContext.themeOverride,
+        shadowDomElementProp,
+        shadowRootElementProp,
+        themeOverrideProp,
+      ],
+    );
 
-  const customOdysseyTheme = useMemo(
-    () => themeOverride && createTheme(deepmerge(odysseyTheme, themeOverride)),
-    [odysseyTheme, themeOverride],
-  );
+    const {
+      designTokensOverride,
+      shadowDomElement,
+      shadowRootElement,
+      themeOverride,
+    } = computedProps;
 
-  return (
-    <MuiThemeProvider theme={customOdysseyTheme ?? odysseyTheme}>
-      <OdysseyDesignTokensContext.Provider value={odysseyTokens}>
-        {children}
-      </OdysseyDesignTokensContext.Provider>
-    </MuiThemeProvider>
-  );
-};
+    const { ref, contrastMode } = useContrastMode(explicitContrastMode);
+
+    const odysseyTokens = useMemo(
+      () => ({ ...Tokens, ...designTokensOverride }),
+      [designTokensOverride],
+    );
+
+    const odysseyTheme = useMemo(
+      () =>
+        createOdysseyMuiTheme({
+          contrastMode,
+          odysseyTokens,
+          shadowRootElement: shadowRootElement || shadowDomElement,
+        }),
+      [contrastMode, odysseyTokens, shadowDomElement, shadowRootElement],
+    );
 
-const MemoizedOdysseyThemeProvider = memo(OdysseyThemeProvider);
+    const customOdysseyTheme = useMemo(
+      () =>
+        createTheme(
+          deepmerge(odysseyTheme, {
+            odysseyContrastMode: contrastMode,
+            ...(themeOverride || {}),
+          } as ThemeOptions),
+        ),
+      [odysseyTheme, themeOverride, contrastMode],
+    );
```

Comment:
This didn't have `as ThemeOptions` in the old code right? I think we don't need it. Unless you're trying to say `{}` is `as ThemeOptions`, but that's separate from making the entire response `ThemeOptions`.

I think this was added for the `contrastMode` prop, but we don't need that anymore.